Improve myths package and API (#5106)
nteract/nteract: packages/myths/README.md

export const store = configureStore({ foo: "bar" });
```

## Epic Definitions

To define epics there are two different shorthand definitions available:

```typescript
export const addToSum =
  iCanAdd.createMyth("addToSum")<number>({

    reduce: (state, action) =>
      state.set("sum", state.get("sum") + action.payload),

    thenDispatch: [
      (action, state) =>
        state.get("sum") - action.payload < 100 && 100 <= state.get("sum")
          ? of(sendNotification.create({message: "Just passed 100!"}))
          : EMPTY,
    ],

    andAlso: [
      {
        // Half the sum every time an error action happens
        when: action => action.error ?? false,
        dispatch: (action, state, addToSum_) =>
          of(addToSum_.create(-state.get("sum") / 2)),
      },
    ],

  });
```

> Use `thenDispatch: []` to define actions then should be generated from actions of the type being defined, and `andAlso: []` to

Reviewer: They are guaranteed to be dispatched to the Redux store, correct? Perhaps it would be helpful to be explicit about this.

```suggestion
Use `thenDispatch: []` to define actions then should be dispatched when actions of the type defined are dispatched, and `andAlso: []` to
```
